awscli/customizations/sso/utils.py

```diff
@@ -65,6 +65,25 @@ def open_browser_with_original_ld_path(url):
         webbrowser.open_new_tab(url)
 
 
+class PrintOnlyHandler(object):
+    def __init__(self, outfile=None):
+        self._outfile = outfile
+
+    def __call__(self, userCode, verificationUri,
+                 verificationUriComplete, **kwargs):
+        opening_msg = (
+            f'Browser will not be automatically opened.\n'
+            f'Please visit the following URL:\n'
+            f'\n{verificationUri}\n'
+            f'\nThen enter the code:\n'
+            f'\n{userCode}\n'
+            f'\nAlternatively, you may visit the following URL:'
+            f'\n{verificationUriComplete}\n'
+            f'\nWhich will autofill the code upon loading.\n'
```

Comment:
It may be nice to move this line up to with the line `Alternatively, you may visit the following URL:`. When trying out the parameter, it took me a few looks to realize that the autofill was referring to the URL and not something that was about to be autofilled in the terminal as login progress was halted at that line.